### PR TITLE
allow setting mutability

### PR DESCRIPTION
**Pull request by athurn**
Adding an option to set the mutability for the repository. This is a backwards compatible change, omitting the option will not change the existing configuration.

### CODE REVIEW

**Comment by 72636c** [comment]
Thanks for the PR @athurn!

Do you think it would make more sense to align with the AWS [API](https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_PutImageTagMutability.html#API_PutImageTagMutability_RequestSyntax) and [CLI](https://docs.aws.amazon.com/cli/latest/reference/ecr/put-image-tag-mutability.html) here?

```yaml
image-tag-mutability: IMMUTABLE
```

File: plugin.yml [hunk 1]
```diff
@@ -15,4 +15,6 @@ configuration:
       type: string
     repository-tags-file:
       type: string
+    image-tag-imutable:
```
Comment (faizhasan) [inline plugin.yml]:
```suggestion
    image-tag-immutable:
```

File: hooks/pre-command [hunk 2]
```diff
@@ -75,6 +75,21 @@ upsert_ecr() {
       --image-scanning-configuration \
       scanOnPush=false
   fi
+
+
+  if [[ "${BUILDKITE_PLUGIN_CREATE_ECR_IMAGE_TAG_IMUTABLE:-}" =~ ^(true|on|1)$ ]]; then
```
Comment (faizhasan) [inline hooks/pre-command]:
```suggestion
  if [[ "${BUILDKITE_PLUGIN_CREATE_ECR_IMAGE_TAG_IMMUTABLE:-}" =~ ^(true|on|1)$ ]]; then
```

File: hooks/pre-command [hunk 3]
```diff
@@ -75,6 +75,21 @@ upsert_ecr() {
...
+    echo '--- Setting ECR image tag mutability to IMMUTABLE'
+    aws ecr put-image-tag-mutability \
+      --repository-name "${repository_name}" \
+      --region "${region}" \
+      --image-tag-mutability "IMMUTABLE"
+  elif [[ "${BUILDKITE_PLUGIN_CREATE_ECR_IMAGE_TAG_IMUTABLE:-}" =~ ^(false|off|0)$ ]]; then
```
Comment (faizhasan) [inline hooks/pre-command]:
```suggestion
  elif [[ "${BUILDKITE_PLUGIN_CREATE_ECR_IMAGE_TAG_IMMUTABLE:-}" =~ ^(false|off|0)$ ]]; then
```

**Comment by athurn** [comment]
@72636c @faizhasan Thank you for your review! Sorry, I totally forgot about this PR. 
I've applied your suggestions, could you please take another look? Thanks.

**Review by gitlon — APPROVED** [review]

**Comment by gitlon** [comment]
@athurn thanks for this change and your patience.

I made some minor amendments directly into your branch and ran tests against these, as that seemed fastest to facilitate release.

- added a test of the input against the enum values
- added a type (string) to the enum schema definition, to match JSON Schema specification (enums should have a type)
- added region to logging header to match other headers (from a PR recently merged to master).

**Pull request merged**